File: Aulas1e2/CD_Java/Java/Sockets/02-Calc/Server/Complexo.py
class Complexo:
    def __init__(self, real=0, imaginary=0):
        self.real = real
        self.imaginary = imaginary

    def __str__(self):
        if self.imaginary != 0:
            return f"{self.real} + {self.imaginary}i"
        else:
            return f"{self.real}"

    def getReal(self):
        return self.real

    def getImaginary(self):
        return self.imaginary

    def addTo(self, number: Complexo) -> Complexo:
        return Complexo(self.real + number.getReal(), self.imaginary + number.getImaginary())

    def subtractTo(self, number: Complexo) -> Complexo:
        return Complexo(self.real - number.getReal(), self.imaginary - number.getImaginary())

    def multiplyBy(self, number: Complexo) -> Complexo:
        return Complexo((self.real*number.getReal() - self.imaginary*number.getImaginary()), (self.real*number.getImaginary() + self.imaginary*number.getReal()))

File: Aulas1e2/CD_Java/Java/Sockets/02-Calc/Client/src/main/java/enidh/cd/sockets/calc/client/Complexo.java
package enidh.cd.sockets.calc.client;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Complexo {

    private final int real;
    private final int imaginary;

    public Complexo() {this(0,0);}

    @JsonCreator
    public Complexo(@JsonProperty("real") int real, @JsonProperty("imaginary") int imaginary){
        this.real = real;
        this.imaginary = imaginary;
    }

    public String toString(){
        return "" + this.real + " + " + this.imaginary + "i";
    }

    public int getReal(){
        return this.real;
    }

    public int getImaginary(){
        return this.imaginary;
    }

    public Complexo addTo(Complexo number){
        return new Complexo(this.real + number.getReal(), this.imaginary + number.getImaginary());
    }

    public Complexo subtractTo(Complexo number){
        return new Complexo(this.real - number.getReal(), this.imaginary - number.getImaginary());
    }

    public Complexo multiplyBy(Complexo number){
        return new Complexo((this.real*number.getReal() - this.imaginary* number.getImaginary()), (this.real*number.getImaginary() + this.imaginary* number.getReal()));
    }
}
